manager/controllers/app/fybrikapplication_controller.go

```diff
@@ -72,18 +73,27 @@ func (r *FybrikApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	observedStatus := applicationContext.Status.DeepCopy()
 	appVersion := applicationContext.GetGeneration()
 
-	// check if observed generation is not the same as appVersion
-	if observedStatus.ObservedGeneration != appVersion {
+	// check if webhooks are enabled and application has been validated before or if validated application is outdated
+	if os.Getenv("ENABLE_WEBHOOKS") == "false" && (string(applicationContext.Status.ValidApplication) == "" || observedStatus.ValidatedGeneration != appVersion) {
```

Comment:
Better make it` != "true"` to capture undefined variables as well

Reply:
Updated in the latest commit